dg-spark/src/main/code/org/finra/datagenerator/RandomNumberEngine.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2014 DataGenerator Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.finra.datagenerator
+
+import java.io.InputStream
+import java.util
+
+import org.finra.datagenerator.distributor.SearchDistributor
+import org.finra.datagenerator.engine.{Engine, Frontier}
+
+import scala.io
+
+/**
+ * Engine Implementation for generating data
+ *
+ * Created by Brijesh on 5/26/2015.
+ */
+
+class RandomNumberEngine extends Engine {
+
+  var totalCount: Int = 0
```

Comment:
You have vals and vars in some of your classes -- they're all public by default. I think you might want to review because some of them are probably best to keep private or protected.
Also, if someone tries to access a val or var from Java, it will look like totalCount() in this case, and if they try to set it, it will look like totalCount_$eq(val), so any public vals or vars would be good to have @BeanProperty on if it's possible they might be used by Java code.
In this case I'm probably overthinking it because it's just an example.